homework6/oop_2.py

```diff
@@ -0,0 +1,200 @@
+
+
+class Person:
+    """
+    The class collect information
+
+    Attributes:
+        last_name (str): Last name
+        first_name (str): First name
+    """
+
+    def __init__(self, last_name: str, first_name: str):
+        self.last_name = last_name
+        self.first_name = first_name
+
+
+class Homework:
+    """
+    The Homework class collect information about task and
+    checking is it complete the task in time
+
+    Attributes:
+        text (str): description a task
+        deadline (int): amount of days for work
+        created (class datetime): current data and time of creation a task
+    """
+
+    def __init__(self, text: str, deadline: int):
+        self.text = text
+        self.deadline = datetime.timedelta(days=deadline)
+        self.created = datetime.datetime.now()
+
+    def is_active(self) -> bool:
+        """Checking if task in active, return True/False"""
+        dead_line = self.deadline + datetime.datetime.now()
+        if dead_line > datetime.datetime.now():
+            return True
+        else:
+            return False
+
+
+class DeadlineError(Exception):
+    """Custom exception class for checking deadlines"""
+
+    def __init__(self, text):
+        self.text = text
+
+
+class Student(Person):
+    """
+    The class collect information about student
+
+    Attributes:
+        last_name (str): Last name of student
+        first_name (str): First name of student
+    """
+
+    def do_homework(self, homework: object, res: str) -> Optional:
+        """Return object "homework" if work overdue"""
+        try:
+            if homework.is_active():
+                return HomeworkResult(self, homework, res)
+            else:
+                raise DeadlineError("You are late")
+        except DeadlineError:
+            raise DeadlineError("You are late")
+
+
+class HomeworkResult:
+    """Class for collecting information about solution of homework"""
+
+    def __init__(self, student, homework: object, solution: str):
+        self.homework = homework
+        self.solution = solution
+        self.author = student
+        self.created = homework.created
+
+
+class Teacher(Person):
+    """
+    The class collect information about teacher
+        Class attributes:
+            homework_done (defaultdict): collect unique value of checked homeworks
+        Attributes:
+            last_name (str): Last name of student
+            first_name (str): First name of student
+    """
+
+    homework_done = defaultdict(set)
+
+    @staticmethod
+    def create_homework(text: str, days: int) -> object:
+        """Create an instance of Homework"""
+        return Homework(text, days)
+
+    def check_homework(self, homeworkresult: object) -> Optional:
+        """Checking solution of homework, if ok -> add to homework_done else: False"""
+        str_solution = homeworkresult.solution
+        if len(str_solution) > 5:
+            self.homework_done[homeworkresult.homework] = str_solution
```

Comment:
homework_done is defaultdict(set). here you just assign str_solution to it. are you sure?

Reply:
also, this gurantees that homeworks solution would be uniwue only for this instance (because of self.). You need to do it unique for the whole (in task: "Общий для всех учителей.")

Reply:
it also should return True in some cases. check task please